elftools/construct/lib/binary.py

```diff
@@ -15,13 +15,7 @@ def int_to_bin(number, width=32):
     """
     if number < 0:
         number += 1 << width
-    i = width - 1
-    bits = bytearray(width)
-    while number and i >= 0:
-        bits[i] = number & 1
-        number >>= 1
-        i -= 1
-    return bytes(bits)
+    return bytes(bool(number & (1 << (width - 1 - bit))) for bit in range(width))
```

Comment:
Same comment as previous PR: please don't make code changes like this to construct, for reasons your PR header explains - I do hope to use it as a dependency eventually

Reply:
I have dropped those code cleanups for now; only the typing related changes should remain.